### PR TITLE
Archive Viewer file import and export

### PR DESCRIPTION
**Pull request by zdomke**
Create a tool to manage importing, exporting, and converting Archive Viewer config files. The tool works as a standalone CLI tool as well as an importable module.

The import & export capabilities are straight forward: given a filename either read in the data or write saved data to the file. The imported file can be of 2 types (.xml and .pyav), but the file extension is ignored if it is in the right format. The tool allows for converting from Java-based Archive Viewer config files to PyDM-based Archive Viewer files.

This Pull Request also implements importing/exporting into the Archive Viewer, making use of the tool described above. File IO is attached to 2 buttons on the application marked "Test Import" and "Test Export". This functionality should be moved eventually.

Changes to the .gitignore to ignore save files (while testing).

### CODE REVIEW

File: archive_viewer/av_file_convert.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+    logger = logging.getLogger("")
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter("[%(asctime)s] [%(levelname)-8s] - %(message)s")
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    logger.setLevel("INFO")
+    handler.setLevel("INFO")
+
+
+class ArchiveViewerFileConverter():
```
Comment (YektaY) [inline archive_viewer/av_file_convert.py:27]: would it be possible to add a class description? thanks

File: archive_viewer/mixins/file_io.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+from pathlib import Path
+from datetime import (datetime, timedelta)
+from urllib.parse import urlparse
+from qtpy.QtWidgets import (QMessageBox, QFileDialog)
+from config import (logger, save_file_dir)
+from av_file_convert import ArchiveViewerFileConverter
+
+
+class FileIOMixin:
+    def file_io_init(self):
```
Comment (YektaY) [inline archive_viewer/mixins/file_io.py:13]: can you add documentation to this method?

**Review by YektaY — COMMENTED** [review]
This looks good. I have a couple of small requests for additional documentation.

File: archive_viewer/av_file_convert.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+        if not srgb:
+            return QColor()
+        elif srgb[0] != '#':
+            rgb_int = int(srgb) & 0xFFFFFFFF
+            srgb = f"#{rgb_int:08X}"
+        return QColor(srgb)
+
+    @staticmethod
+    def remove_null_values(dict_in: Dict) -> Dict:
+        """Remove all key-value pairs from a given dictionary where the value is None"""
```
Comment (zdomke) [inline archive_viewer/av_file_convert.py:298]: @YektaY do you want me to add the parameters & returns to this docstring?
It feels like a pretty straightforward helper function, but I can add more to the docs to match styling

Comment (YektaY) [reply archive_viewer/av_file_convert.py:298]: adding more would be helpful for the auto documentation we will have for the website

File: archive_viewer/av_file_convert.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        for formula_in in data_in['formula']:
+            # TODO convert formulas once formulas are implemented for ArchiveViewer
+            pass
+
+        self.stored_data = converted_data
+        return self.stored_data
+
+    @classmethod
+    def reformat_date(cls, input_str: str) -> str:
+        """Convert a time string from the format 'MM/DD/YYYY' --> 'YYYY-MM-DD'
```
[the remark this answers is not in the document]
Comment (zdomke) [reply archive_viewer/av_file_convert.py:198]: Solved in commit 3f00fff

File: archive_viewer/archive_viewer.py [hunk 5]
```diff
@@ -35,6 +36,11 @@ def __init__(self, parent=None, args=None, macros=None, ui_filename=__file__.rep
         app = QApplication.instance()
         app.setStyle(CenterCheckStyle())
 
+    def file_menu_items(self):
```
[the remark this answers is not in the document]
Comment (zdomke) [reply archive_viewer/archive_viewer.py]: Solved in commit 3f00fff13322325cd1f81101ea9b124ff61f02fb

File: archive_viewer/av_file_convert.py (hunk 1 above)
Comment (zdomke) [reply archive_viewer/av_file_convert.py:27]: Solved in commit 3f00fff13322325cd1f81101ea9b124ff61f02fb

File: archive_viewer/mixins/file_io.py (hunk 2 above)
Comment (zdomke) [reply archive_viewer/mixins/file_io.py]: Solved in commit 3f00fff13322325cd1f81101ea9b124ff61f02fb

File: archive_viewer/av_file_convert.py (hunk 3 above)
Comment (zdomke) [reply archive_viewer/av_file_convert.py]: Solved in commit 3f00fff13322325cd1f81101ea9b124ff61f02fb

**Review by YektaY — APPROVED** [review]

**Pull request merged**